app/controllers/bookmarks_controller.rb

```diff
@@ -132,23 +132,27 @@ def status_action documents
     status = params['action']
     Array(documents.map(&:id)).each do |id|
       media_object = MediaObject.find(id)
-      if cannot? :update, media_object
-        errors += ["#{media_object.title} (#{id}) #{t('blacklight.messages.permission_denied')}."]
+      sucess = false
+      case status
```

Comment:
This logic needs refactoring? Forcing a switch statement and a flag when a simple if else could do?

```ruby 

if status.publish || status.unpublish
  success_ids << id if can?(:update, media_object) || can?(:unpublish, media_object)
else
  errors << #{media_object.title} (#{id}) #{t('blacklight.messages.permission_denied')}."
end
```